### DISCUSSION TITLE
Can a mesh be "too fine"?

### DISCUSSION THREAD

**Discussion by xshang93** [neperfepx/FEPX#72] [Q&A] [answered]
This might be a silly question but I wonder if a mesh can ever be "too fine", only from convergency perspective? I'm recently running some fepx simulations but I found finer mesh at ~100k+ is not converging but at ~35k it is converging. I couldn't justify why this is happening, could anyone give any insight?

PS. It could be that I'm drawing this conclusion wrongly, but I wanted to gather some ideas first.

TIA!

**Comment by mkasemer** [comment, marked as answer]
@xshang93 - In general this shouldn't be a problem, at least within the limits you are proposing. 100k elements should be relatively routine.

Convergence issues can arise for lots of reasons. Perhaps boundary conditions are not restrictive enough, or you have a mesh with poor-quality elements. The latter _can_ arise due to mesh refinement, if done improperly. But no, in general you should not exhibit convergence issues with "too fine a mesh", especially considering only 100k elements.

There is another option that there could be issues with your compiler / system. I seem to recall you have issues on the HPC. Have you independently confirmed this mesh issue on a different computer using a different compiler? I recommend Ubuntu, using Gfortran, which is what FEPX is currently tested on.

**Reply by xshang93** [reply]
Hi @mkasemer , thanks for your response. This aligns with my understanding towards FEA. There must be something I missed, and I will try to dig it out.

In regarding to the HPC issue, I'm pretty confident that it's working properly for the moment - at lease when I use one single node for computation. Most of my computations so far converges when meshed properly.

Thanks,
Xiao
